Fix hostresolver setting for VZ, but disable it by default

jandubois opened the pull request:
Since it was implicitly disabled for VZ, we cannot simply fix the bug, but change the default to match the previous de-facto default.

This PR now disables HostResolver by default, except for QEMU.

This implements option (1) from #4089.

AkihiroSuda commented on templates/default.yaml:
```diff
@@ -565,7 +565,7 @@ propagateProxyEnv: null
 # IPv6 configured as well.
 hostResolver:
-  # 🟢 Builtin default: true
+  # 🟢 Builtin default: true (except for vmType: vz, where it is false)
```
https://lima-vm.io/docs/config/network/user/#dns-19216853 needs to be updated to explain why this is not needed on vz?

AkihiroSuda replied:
Alternatively we can make it false by default, except for qemu?

AkihiroSuda commented on website/content/en/docs/config/network/user.md:
```diff
@@ -43,4 +43,6 @@ DNS over tcp is rarely used. It is usually only used either when user explicitly
-If `hostResolver.enabled` is false, then DNS servers can be configured manually in `lima.yaml` via the `dns` setting. If that list is empty, then Lima will either use the slirp DNS (on Linux), or the nameservers from the first host interface in service order that has an assigned IPv4 address (on macOS).
+If `hostResolver.enabled` is false (default for all drivers except `QEMU`), then DNS servers can be configured manually in `lima.yaml` via the `dns` setting. If that list is empty, then Lima will either use the slirp DNS (on Linux), or the nameservers from the first host interface in service order that has an assigned IPv4 address (on macOS).
+
+The host resolver is not needed with the `VZ` driver because the `gvisor-tap-vsock` implementation used by `usernet` has the same functionality built-in.
```
nit

```suggestion
The host resolver is not needed with the `vz` driver because the `gvisor-tap-vsock` implementation used by `usernet` has the same functionality built-in.
```

or

```suggestion
The host resolver is not needed with the VZ driver because the `gvisor-tap-vsock` implementation used by `usernet` has the same functionality built-in.
```

Same for `QEMU`

jandubois committed:
Fix hostresolver setting for VZ, but disable it by default

Since it was implicitly disabled for VZ, we cannot simply fix the bug,
but change the default to match the previous de-facto default.

See https://github.com/lima-vm/lima/issues/4089 for more info.

Signed-off-by: Jan Dubois <jan.dubois@suse.com>

jandubois commented:
> It does pass for me locally, fwiw.

I can reproduce the problem even on macOS when explicitly asking for qemu:

```
limactl start -y --vm-type qemu ./hack/test-templates/net-user-v2.yaml
```

It seems like usernet configuration is different between QEMU and VZ.
